placementinsql: add the survival preferences docs (#12428)
pingcap/docs: placement-rules-in-sql.md

| `FOLLOWER_CONSTRAINTS`     | A list of constraints that only apply to followers.                                   |
| `LEARNER_CONSTRAINTS`      | A list of constraints that only apply to learners.                                     |
| `LEARNERS`                 | The number of learners. |

> | `SURVIVAL_PREFERENCE`      | The replica placement priority according to the label of the disaster tolerance level. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |

Author: ```suggestion
| `SURVIVAL_PREFERENCE`      | Specifies the replica placement priority according to the disaster tolerance level of the labels. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |
```

> | `SURVIVAL_PREFERENCE`      | Specifies the replica placement priority according to the disaster tolerance level of the labels. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |

Reviewer: ```suggestion
| `SURVIVAL_PREFERENCE`      | The replica placement priority according to the disaster tolerance level of the labels. For example, `SURVIVAL_PREFERENCE="[region, zone, host]"`. |
```

>
> Dictionary and list formats are based on the YAML parser, but the YAML syntax might be incorrectly parsed. For example, `"{+disk=ssd:1,+disk=nvme:2}"` is incorrectly parsed as `'{"+disk=ssd:1": null, "+disk=nvme:1": null}'`. But `"{+disk=ssd: 1,+disk=nvme: 1}"` is correctly parsed as `'{"+disk=ssd": 1, "+disk=nvme": 1}'`.

> ### Survival Preferences

Author: ```suggestion
### Survival preferences
```

For some important data, you might want to have multiple replicas across availability zones to get high disaster survivability, such as survivability at the cloud region level. To achieve that, you can set `SURVIVAL_PREFERENCES` for your data.

> For example, you can set the `SURVIVAL_PREFERENCES` as follows to require the data to meet the configured survival preferences as much as possible:

Author: ```suggestion
For example, when creating a placement policy, you can set the `SURVIVAL_PREFERENCES` as follows to require the data to meet the configured survival preferences as much as possible:
```

``` sql
CREATE PLACEMENT POLICY multiregion
    follower=4

>     PRIMARY_REGION="region1"

Reviewer: ```suggestion
    PRIMARY_REGION="us-east-1"
```

    SURVIVAL_PREFERENCES="[region, zone]";
```

> Then, for tables with this policy, data will first meet the survival objectives of data isolation across regions, and then ensure the survival objectives of data isolation across availability zones.

Author: ```suggestion
Then, for tables attached with this policy, data will first meet the survival objectives of data isolation across regions, and then meet the survival objectives of data isolation across availability zones.
```

> Then, for tables attached with this policy, data will first meet the survival objectives of data isolation across regions, and then meet the survival objectives of data isolation across availability zones.

Reviewer: ```suggestion
Then, for the tables attached with this policy, the data will first meet the survival objectives of data isolation across regions, and then meet the survival objectives of data isolation across availability zones.
```

> **Note:**
>

> > `SURVIVAL_PREFERENCES` is equivalent to `LOCATION_LABELS` in PD. For more information, see [Schedule replicas by topology labels](/schedule-replicas-by-topology-labels.md).

Reviewer: ```suggestion
> `SURVIVAL_PREFERENCES` is equivalent to `LOCATION_LABELS` in PD. For more information, see [Schedule Replicas by Topology Labels](/schedule-replicas-by-topology-labels.md).
```

>
> Dictionary and list formats are based on the YAML parser, but the YAML syntax might be incorrectly parsed. For example, `"{+disk=ssd:1,+disk=nvme:2}"` is incorrectly parsed as `'{"+disk=ssd:1": null, "+disk=nvme:1": null}'`. But `"{+disk=ssd: 1,+disk=nvme: 1}"` is correctly parsed as `'{"+disk=ssd": 1, "+disk=nvme": 1}'`.

### Survival preferences

When you create or modify a placement policy, you can use the `SURVIVAL_PREFERENCES` option to set the preferred survivability for your data.

For example, assuming that you have a TiDB cluster across 3 availability zones, with multiple TiKV instances deployed on each host in each zone. And when creating placement policies for this cluster, you have set the `SURVIVAL_PREFERENCES` as follows:

``` sql
CREATE PLACEMENT POLICY multiaz SURVIVAL_PREFERENCES="[zone, host]";
CREATE PLACEMENT POLICY singleaz CONSTRAINTS="[+zone=zone1]" SURVIVAL_PREFERENCES="[host]";
```

After creating the placement policies, you can attach them to the corresponding tables as needed:

> - For tables attached with the `multiaz` placement policy, data will be placed in 3 replicas in different availability zones, prioritizing survival goals of data isolation cross zones, followed by survival goals of data isolation cross hosts.

Reviewer: 原文是 绑定，用 attached 还是用 bound?

Author: 这篇中文文档里提到了多次“绑定”，对应的英文里统一用了 attach （https://docs.pingcap.com/tidb/dev/placement-rules-in-sql），所以这里用了 attach
